### PR TITLE
Get rid of some unneeded transmutes and other special cases for enum/…

### PR DESCRIPTION
**Pull request by sdroege**
…flags properties

They implement the Value traits now, no special handling needed here.

### CODE REVIEW

**Comment by EPashkin** [comment]
This [runtime](https://github.com/EPashkin/gtk-rs-runtime-tests/blob/master/tests/child_properties.rs#L25-L28) test is failing (at minimum on Windows)

**Comment by EPashkin** [comment]
But setters seems work, at minimum change https://github.com/gtk-rs/examples/blob/master/src/bin/child-properties.rs#L48 give effect.

**Comment by sdroege** [comment]
How does the child-properties example fail for you? It works just fine here on Linux
If it fails, that would suggest that the `Value` traits implementations for enums is broken generally on Windows.

**Comment by sdroege** [comment]
Ah, not that example fails for you but when using the getters. Yes, I can confirm that!

**Comment by sdroege** [comment]
The returned `Value` is of type `gint` instead of `GtkPackType`.

**Comment by sdroege** [comment]
The problem is https://github.com/gtk-rs/gir/blob/master/src/codegen/property_body.rs#L123 . This is wrong even before my change, and if you compile GObject with further runtime checks it will also cause errors.

This should instead use something like
```
        let mut value = Value::uninitialized();
        gobject_ffi::g_value_init(value.to_glib_none_mut().0, gtk::PackType::static_type().to_glib());
```

**Comment by EPashkin** [comment]
Agree, and then calculating `default_value` need removed too.
Not sure that it can be normally generated :(

**Comment by sdroege** [comment]
It can, yes

**Comment by sdroege** [comment]
Fixed, related to the `Vec<String>` special case here there's https://github.com/gtk-rs/glib/pull/264

**Comment by sdroege** [comment]
It is however *not* a dependency for this PR.

**Comment by EPashkin** [comment]
@sdroege At first glance it looks fine, I check at evening.

**Comment by sdroege** [comment]
Thanks!

File: src/codegen/properties.rs [hunk 1]
```diff
@@ -127,19 +127,19 @@ fn declaration(env: &Env, prop: &Property) -> String {
...
-        .conversion(prop.conversion);
-    if let Some(ref default_value) = prop.default_value {
-        builder.default_value(default_value);
+        .is_nullable(*prop.nullable);
+
+    if let Ok(type_) = rust_type(env, prop.typ) {
+        builder.type_(&type_);
     } else {
-        builder.default_value("/*Unknown default value*/");
+        builder.type_("/*Unknown type */");
```
Comment (EPashkin) [inline src/codegen/properties.rs]: IMHO unneeded space at end of string.

File: src/codegen/child_properties.rs [hunk 2]
```diff
@@ -128,12 +128,12 @@ fn body(prop: &ChildProperty, is_get: bool) -> Chunk {
...
-        .conversion(prop.conversion);
-    if let Some(ref default_value) = prop.default_value {
-        builder.default_value(default_value);
+        .is_into(prop.is_into);
+
+    if let Ok(type_) = rust_type(env, prop.typ) {
+        builder.type_(&type_);
     } else {
-        builder.default_value("/*Unknown default value*/");
+        builder.type_("/*Unknown type */");
```
Comment (EPashkin) [inline src/codegen/child_properties.rs]: Extra space here too.

**Comment by EPashkin** [comment]
@sdroege Runtime test now passed, only small nit last.

**Comment by EPashkin** [comment]
Hm, there many `Value::uninitialized/g_value_init` maybe we need add `Value::withType` at some time

File: src/codegen/child_properties.rs (hunk 2 above)
Comment (sdroege) [reply src/codegen/child_properties.rs]: Removed the two spaces :)

**Comment by sdroege** [comment]
> Hm, there many Value::uninitialized/g_value_init maybe we need add Value::with_type at some time

That could make sense, but I'd prefer to wait with that until after everything is re-gen'd. This here should ideally go into the next bugfix release as it has the potential to blow up things.

**Comment by EPashkin** [comment]
@sdroege Thanks

**Comment by EPashkin** [comment]
By way, I tried to use `ObjectExt::get_property` in `get_property_xxx`.
Strange that it need  `glib::value::SetValue` constraint on impl,
and as it always check property existence,
seems current version is better.

**Comment by sdroege** [comment]
We could probably get rid of that trait bound somehow, and also add an unsafe variant that does not check for the property's existence... But there's not much difference in the code overall then. Unlike with signal connectors :)

**Comment by sdroege** [comment]
I'll send regent PRs for everything later unless someone else wants too

**Comment by EPashkin** [comment]
About `SetValue`: as this constraint seems need only for one function we can extract it to separate trait.

I have free time so I add PR for gtk at minimum.

**Comment by sdroege** [comment]
Thanks, all looking good to me

**Pull request merged**